modules/dir2000/src/test/java/com/hillel/elementary/javageeks/dir/homework5/fraction/FractionTest.java

```diff
@@ -0,0 +1,79 @@
+package com.hillel.elementary.javageeks.dir.homework5.fraction;
+
+import org.junit.jupiter.api.Test;
+
+import java.util.Iterator;
+import java.util.LinkedList;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class FractionTest {
+  @Test
+  public void shouldThrowArithmeticExceptionOnCreationWithZeroDenominator(){
+    //given
+    Fraction fraction = null;
+
+    //when
+    try {
+      fraction = new Fraction(8, 0);
+    } catch (ArithmeticException e) {
+    }
+
+    //then
+    assertThat(fraction).isNull();
+  }
+
+  @Test
+  public void shouldThrowArithmeticExceptionOnSettingZeroDenominator(){
+    //given
+    Fraction fraction = new Fraction(8, 3);
+
+    //when
+    try {
+      fraction.setDenominator(0);
```

Comment:
We don't use try catch blocks in our tests. Use assertThatThrownBy. But don't worry to much if it looks complicated.

Reply:
I read a little about lambda expressions)